### PR TITLE
Add tracing config

### PR DESCRIPTION
**Pull request by andrewmains12**
This is a small preparatory diff to add support for tracing to m3query. I figured I might as well do the config here, since that's where we have metrics + logging. Let me know if you'd rather have this code elsewhere!

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -47,6 +47,14 @@ import:
...
   - cmp
 
+- package: github.com/uber/jaeger-lib
+  version: ^1.5.0
+
+- package: github.com/uber/jaeger-client-go
+  version: ^2.7.0
+
+
+
```
Comment (benraskin92) [inline glide.yaml]: Remove some of these lines?

File: instrument/tracing.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	cfg.Jaeger.ServiceName = serviceName
+	jaegerLog := jaegerzap.NewLogger(logger)
+	tracer, jaegerCloser, err := cfg.Jaeger.NewTracer(
+		jaegercfg.Logger(jaegerLog),
+		jaegercfg.Metrics(jaegertally.Wrap(scope)))
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize jaeger: %s", err.Error())
+	}
```
Comment (benraskin92) [inline instrument/tracing.go]: newline

File: instrument/tracing.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+
+func (cfg *TracingConfiguration) NewTracer(serviceName string, scope tally.Scope, logger *zap.Logger) (opentracing.Tracer, io.Closer, error) {
+	if cfg.Backend == "" {
+		return opentracing.NoopTracer{}, noopCloser{}, nil
+	}
+
+	if cfg.Backend != TracingBackendJaeger {
+		return nil, nil, fmt.Errorf("unknown tracing backend: %s", cfg.Backend)
```
Comment (benraskin92) [inline instrument/tracing.go]: Maybe add that `jaeger` is the only supported backend?

File: instrument/tracing.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	cfg.Jaeger.ServiceName = serviceName
```
Comment (benraskin92) [inline instrument/tracing.go]: Check to make sure `cfg.Jaeger.ServiceName != ""`

File: glide.yaml (hunk 1 above)
Comment (andrewmains12) [reply glide.yaml]: Whoops yeah will do.

File: instrument/tracing.go (hunk 3 above)
Comment (andrewmains12) [reply instrument/tracing.go]: Good call!

File: instrument/tracing.go (hunk 4 above)
Comment (andrewmains12) [reply instrument/tracing.go]: Good call.

**Pull request merged**